IMPORT

In [1]:
import pandas as pd
import os

EXAMPLE DATASET

In [2]:
dataset = pd.read_csv(r'C:\Users\yhcel\Desktop\code\anaconda-navigator\csv-files\breast-cancer.csv')

In [3]:
dataset.shape

(569, 31)

In [4]:
print(dataset.head())

  diagnosis  radius_mean  texture_mean  perimeter_mean  area_mean  \
0         M     1.097064     -2.073335        1.269934   0.984375   
1         M     1.829821     -0.353632        1.685955   1.908708   
2         M     1.579888      0.456187        1.566503   1.558884   
3         M    -0.768909      0.253732       -0.592687  -0.764464   
4         M     1.750297     -1.151816        1.776573   1.826229   

   smoothness_mean  compactness_mean  concavity_mean  concave points_mean  \
0         1.568466          3.283515        2.652874             2.532475   
1        -0.826962         -0.487072       -0.023846             0.548144   
2         0.942210          1.052926        1.363478             2.037231   
3         3.283553          3.402909        1.915897             1.451707   
4         0.280372          0.539340        1.371011             1.428493   

   symmetry_mean  ...  radius_worst  texture_worst  perimeter_worst  \
0       2.217515  ...      1.886690      -1.359293 

LABEL ENCODING

In [5]:
from sklearn.preprocessing import LabelEncoder

In [6]:
labelencoder = LabelEncoder()

In [7]:
dataset['diagnosis'] = labelencoder.fit_transform(dataset['diagnosis'].values)

TRAIN TEST SPLIT

In [8]:
from sklearn.model_selection import train_test_split

In [9]:
train, test = train_test_split(dataset, test_size = 0.3)

In [10]:
x_train = train.drop('diagnosis', axis=1)
y_train = train.loc[:, 'diagnosis']

In [11]:
x_test = test.drop('diagnosis', axis=1)
y_test = test.loc[:, 'diagnosis']

LOGISTIC REGRESSION

In [12]:
from sklearn.linear_model import LogisticRegression

In [13]:
model = LogisticRegression()

In [14]:
print(model.fit(x_train, y_train))

LogisticRegression()


In [15]:
predictions = model.predict(x_test)
print(predictions)

[0 0 0 0 1 1 1 0 0 0 1 1 0 0 0 0 1 0 0 1 1 0 0 1 1 1 0 1 0 1 1 0 1 0 0 0 1
 1 0 1 0 1 0 0 0 1 1 0 0 1 0 0 0 1 1 0 0 1 0 1 1 1 0 0 0 0 1 0 1 1 1 0 1 1
 0 0 0 0 0 0 0 1 1 0 1 0 1 0 1 0 1 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1
 1 0 1 0 0 0 1 0 0 1 1 0 1 0 1 0 0 1 1 0 0 1 0 0 1 1 0 0 1 0 1 1 0 0 0 0 1
 1 1 0 0 1 0 0 1 0 0 0 0 1 1 0 0 0 1 1 1 0 0 0]


CONFUSION MATRIX

In [16]:
from sklearn.metrics import confusion_matrix

In [17]:
confusion_matrix(y_test, predictions)

array([[103,   0],
       [  0,  68]], dtype=int64)

103 = True Negative , 0 = False Positive , 0 = False Negative , 68 = True Positive 

CLASSIFICATION REPORT

In [18]:
from sklearn.metrics import classification_report

In [19]:
print(classification_report(y_test, predictions)) 

              precision    recall  f1-score   support

           0       1.00      1.00      1.00       103
           1       1.00      1.00      1.00        68

    accuracy                           1.00       171
   macro avg       1.00      1.00      1.00       171
weighted avg       1.00      1.00      1.00       171



A support vector machine (SVM) is a type of supervised learning algorithm used in machine learning to solve classification and regression tasks; SVMs are particularly good at solving binary classification problems, which require classifying the elements of a data set into two groups.


Hyper-Space is used for more than 3 features when making classifications.

If hyper-planes are "linear" we use "linear kernel"
If hyper-planes are "non-linear" we use "polynomal kernel"

LINEAR SVC

In [20]:
from sklearn.svm import LinearSVC

In [21]:
model_2 = LinearSVC()

In [23]:
print(model_2.fit(x_train, y_train))

LinearSVC()


C:\Users\yhcel\anaconda3\Lib\site-packages\sklearn\svm\_classes.py:32: FutureWarning: The default value of `dual` will change from `True` to `'auto'` in 1.5. Set the value of `dual` explicitly to suppress the warning.
  warnings.warn(


In [25]:
predictions_2 = model_2.predict(x_test)

In [26]:
print(predictions_2)

[0 0 0 0 1 1 1 0 0 0 1 1 0 0 0 0 1 0 0 1 1 0 0 1 1 1 0 1 0 1 1 0 1 0 0 0 1
 1 0 1 0 1 0 0 0 1 1 0 0 1 0 0 0 1 1 0 0 1 0 1 1 1 0 0 0 0 1 0 1 1 1 0 1 1
 0 0 0 0 0 0 0 1 1 0 1 0 1 0 1 0 1 1 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1
 1 0 1 0 0 0 1 0 0 1 1 0 1 0 1 0 0 1 1 0 0 1 0 0 1 1 0 0 1 1 1 1 0 0 0 0 1
 1 1 0 0 0 0 0 1 0 0 0 0 1 1 0 0 0 1 1 1 0 0 0]


In [27]:
confusion_matrix(y_test, predictions_2)

array([[101,   2],
       [  1,  67]], dtype=int64)

In [28]:
print(classification_report(y_test, predictions_2))

              precision    recall  f1-score   support

           0       0.99      0.98      0.99       103
           1       0.97      0.99      0.98        68

    accuracy                           0.98       171
   macro avg       0.98      0.98      0.98       171
weighted avg       0.98      0.98      0.98       171

